src/precision.jl

```diff
@@ -1,51 +1,40 @@
 const DEFAULT_PRECISION = Ref{Int}(256)
 
-if VERSION < v"1.8.0"
-    # Types
-    Base.precision(::Type{<:Union{ArbTypes,ArbStructTypes,Ptr{<:ArbStructTypes}}}) =
-        DEFAULT_PRECISION[]
-    # Types not storing their precision
-    Base.precision(x::Union{ArbStructTypes,Ptr{<:ArbStructTypes}}) = DEFAULT_PRECISION[]
-    # Types storing their precision
-    Base.precision(x::ArbTypes) = x.prec
-    # Mag doesn't store a precision
-    Base.precision(::MagOrRef) = DEFAULT_PRECISION[]
-    # ArbSeries and AcbSeries don't store their precision directly
-    Base.precision(x::Union{ArbSeries,AcbSeries}) = precision(x.poly)
-else
-    # Since Julia 1.8.0 Base.precision calls Base._precision and by
-    # overloading that we automatically support giving base argument.
+let f
+    f = if VERSION < v"1.8.0-DEV.725"
+        :precision
+    elseif VERSION < v"1.11.0-DEV.1363"
+        # Since Julia 1.8.0 Base.precision calls Base._precision and by
+        # overloading that we automatically support giving base argument.
+        # Ref: https://github.com/JuliaLang/julia/pull/42428
+        :_precision
+    else
+        # Since Julia 1.11.0, the single-argument `Base._precision` is
+        # renamed to `Base._precision_with_base_2`
+        # Ref: https://github.com/JuliaLang/julia/pull/52910
+        :_precision_with_base_2
```

Comment:
The name is prefixed with an underscore to make it clear it's internal. Why trespass!?

Reply:
Trespassing or not ;)

I personally don't have time now to delve into (internals or not) the design of `Base.precision` to figure out what would suit us best and will be least maintenance in the future. If you do have a clear solution we're all ears ;)